proposals/0029-cooperative-vector.md

```diff
@@ -102,383 +113,325 @@ Introduce new HLSL intrinsics to accelarate matrix-vector operations. In this sp
-  FloatE5M2             = 11,
-  Unsupported           = 32
-};
-
-enum class DXILMatrixLayout : uint {
-  RowMajor              = 0,
-  ColumnMajor           = 1,
-  InferencingOptimal    = 2,
-  TrainingOptimal       = 3,
-};
 
-template<uint m, uint k, uint input_interp, uint matrix_interp, uint bias_interp, 
-         uint layout, bool transpose>
-struct VecMatOpDescriptor {
-  static const uint M               = m;
-  static const uint K               = k;
-  static const uint Ii              = input_interp;
-  static const uint Mi              = matrix_interp;
-  static const uint Bi              = bias_interp;
-  static const uint Layout          = layout;
-  static const bool Transposed      = transpose;
-};
+#### Overview
 
-// Result = Matrix * InputVector + Bias
-template<typename DESC, typename InputTy, typename ResultTy, uint InputComponents>
-vector<ResultTy, DESC::M> VectorMatrixMulAdd(vector<InputTy, InputComponents> InputVector,
-                               (RW)ByteAddressBuffer Matrix,
-                               uint MatrixOffset,
-                               uint MatrixStride,
-                               (RW)ByteAddressBuffer BiasVector,
-                               uint BiasOffset);
-
-// Result = Matrix * InputVector
-template<typename DESC, typename InputTy, typename ResultTy, uint InputComponents>
-vector<ResultTy, DESC::M> VectorMatrixMul(vector<InputTy, InputComponents> InputVector,
-                             (RW)ByteAddressBuffer Matrix,
-                             uint MatrixOffset,
-                             uint MatrixStride);
+The `@dx.op.matvecmul` operation multiplies a **MxK** dimension matrix and a
+**K** sized input vector. The matrix is loaded from memory while the vector is
+stored in a variable.
 
-```
+The `@dx.op.matvecmuladd` operation behaves as `@dx.op.matvecmul`, but also adds
+an **M**-sized bias vector (loaded from memory) to the result.
 
-*InputVector* is the vector operand of the matrix-vector mul/mul-add operation. *InputTy* is the physical storage type
- of the elements of the vector, which might vary from the actual type that the elements of the vector are interpreted
- as, *InputInptretation* from *DESC*. *InputComponents* is the number of components in the input vector, which equals
- the matrix dimension *K* for a non-packed type and for a packed type, equals the least number that can hold *K* values
- of the packed type. Where, packed type, refers to types like `SignedInt8x4Packed` where each 32-bit element of the
- vector corresponds to four 8-bit signed integers; Unpacked types are the standard types like float16, uint etc. The
- elements of the *InputVector* are converted to type specified by *DESC: Ii* present in, if it is legal. More details
- in the [Type Interpretations](#type-interpretations) section.
+> Note that the dimensions of the matrix are **M**x**K** versus **M**x**N**
+> usually found in linear algebra textbooks. This is to futureproof for
```

Comment:
Found the answer to my question above. But this sentence is a little weird. Suggestion:
'Note that the dimensions of the matrix are ***** instead of ******, which is usually found in linear algebra textbooks.

Reply:
Thanks - I'll add the "the".